options/posix/generic/unistd-stubs.cpp

```diff
@@ -683,9 +684,13 @@ unsigned long sysconf(int number) {
 		case _SC_PAGE_SIZE:
 			return mlibc::page_size;
 		case _SC_OPEN_MAX:
-			// TODO: actually return a proper value for _SC_OPEN_MAX
-			mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_OPEN_MAX) returns arbitrary value 256\e[39m" << frg::endlog;
-			return 256;
+			struct rlimit ru;
+			if (getrlimit(RLIMIT_NOFILE, &ru) < 0) {
```

Comment:
This function can't fail with anything but ENOSYS, might be worth asserting for that when logging

Reply:
On linux, the man page states that a other errno values can be set, namely `EFAULT` and `EINVAL`.

Reply:
yes, and neither is possible here, unless NOFILE is not implemented